ext/DataInterpolationsChainRulesCoreExt.jl

```diff
@@ -74,6 +74,11 @@ function u_tangent(A::LinearInterpolation, t, Δ)
     out
 end
 
+function _quad_interp_indices(A::QuadraticInterpolation, t::Number, iguess)
```

Comment:
Just to confirm, can't `u_tangent` for QuadraticInterpolation be reformulated using `α` and `β` parameters you added?

Reply:
I'ts possible, but I suspect this is more efficient. Also, that might break AD types that do not support mutation when those parameters are cached?

Reply:
And you still need to have the logic somewhere to see which mode (forward/backward looking interpolation) was used for the `u` gradient.

Reply:
Alright, then we can keep this.